src/ucs/config/parser.c

```diff
@@ -1851,6 +1852,64 @@ void ucs_config_parser_print_all_opts(FILE *stream, const char *prefix,
     }
 }
 
+static void ucs_config_parser_concat_fuzzy_matches_recurs(
+        const ucs_config_field_t *fields, const char *env_prefix,
+        const char *table_prefix, const char *user_var,
+        ucs_string_buffer_t *fuzzy_buff)
+{
+    char ucx_var[128]                 = {0};
+    const uint32_t MAX_FUZZY_DISTANCE = 3;
+    const ucs_config_field_t *field;
+    uint32_t distance;
+
+    for (field = fields; !ucs_config_field_is_last(field); ++field) {
+        if (ucs_config_is_table_field(field)) {
+            ucs_config_parser_concat_fuzzy_matches_recurs(
+                    field->parser.arg, env_prefix, table_prefix, user_var,
+                    fuzzy_buff);
+        }
+
+        snprintf(ucx_var, sizeof(ucx_var), "%s%s%s", env_prefix,
+                 table_prefix ? table_prefix : "", field->name);
+
+        distance = ucs_levenshtein(user_var, ucx_var);
+        if ((distance > 0) && (distance <= MAX_FUZZY_DISTANCE)) {
+            ucs_string_buffer_appendf(fuzzy_buff, "%s, ",
+                                      ucx_var);
+        }
+    }
+}
+
+static void ucs_config_parser_concat_fuzzy_matches(
+        const char *env_prefix, const char *user_var,
+        ucs_string_buffer_t *msg_buff)
+{
+    const ucs_config_global_list_entry_t *entry;
+    ucs_string_buffer_t fuzzy_buff;
+
+    ucs_string_buffer_init(&fuzzy_buff);
+
+    ucs_list_for_each(entry, &ucs_config_global_list, list) {
+        if ((entry->table == NULL) ||
+            (ucs_config_field_is_last(&entry->table[0]))) {
+            continue;
+        }
+
+        ucs_config_parser_concat_fuzzy_matches_recurs(
+                entry->table, env_prefix, entry->prefix, user_var,
+                &fuzzy_buff);
+    }
+
+    if (ucs_string_buffer_length(&fuzzy_buff) > 0) {
+        ucs_string_buffer_rtrim(&fuzzy_buff, ", ");
+        ucs_string_buffer_appendf(msg_buff, " (did you mean: %s?)",
+                                  ucs_string_buffer_cstr(&fuzzy_buff));
+    }
+
+    ucs_string_buffer_appendf(msg_buff, "; ");
```

Comment:
I think placing a separator ("; ") is not a responsibility of this function. This operation should be performed on the place where we append `unused_vars_strb`.